conflation/mapillary.py

```diff
@@ -1,60 +1,106 @@
-from dateutil import parser
-import multiprocessing
+import datetime
+import requests
 import os
 import pickle
-import time
+import multiprocessing
+from dateutil import parser
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
 
-import requests
+from conflation import util
+from conflation import trace_filter
 
-from . import util
+SEQUENCES_PER_PAGE_DEFAULT = 10  # How many sequences to receive on each page of the API call
+IMAGES_PER_PAGE_DEFAULT = 1000  # How many sequences to receive on each page of the API call
+# We only look for sequences beyond the start date, by default a year ago
+SEQUENCE_START_DATE_DEFAULT = (
+    datetime.date.today() - datetime.timedelta(days=365)
+).isoformat()
+SKIP_IF_FEWER_IMAGES_THAN_DEFAULT = (
+    10  # We will skip any sequences if they have fewer than this number of images
+)
+SEQUENCE_URL = "https://a.mapillary.com/v3/sequences_without_images?client_id={}&bbox={}&per_page={}&start_date={}"
+IMAGES_URL = "https://a.mapillary.com/v3/images?client_id={}&sequence_keys={}&per_page={}"
 
-SEQUENCES_PER_PAGE_DEFAULT = 50  # How many sequences to receive on each page of the API call
-SKIP_IF_FEWER_IMAGES_THAN_DEFAULT = 10  # We will skip any sequences if they have fewer than this number of images
-SEQUENCE_URL = 'https://a.mapillary.com/v3/sequences_without_images?client_id={}&bbox={}&per_page={}'
-IMAGES_URL = 'https://a.mapillary.com/v3/images?client_id={}&sequence_keys={}'
 
+def run(
+    bbox: str, output_dir: str, output_tmp_dir: str, traces_source: dict, processes: int
+) -> None:
+    """
+    Entrypoint for pulling trace date from Mapillary APIs. Will pull all trace data in the given bbox and store it in
+    the output_dir, using the number of processes specified and any conf values from traces_source.
 
-def run(bbox: str, traces_source: any, processes: int):
+    :param bbox: Bounding box we are searching over, in the format of 'min_lon,min_lat,max_lon,max_lat'
+    :param output_dir: Dir where trace data will be pickled to
+    :param output_tmp_dir: Dir where temp output files will be stored (should be empty upon completion)
+    :param traces_source: Dict of configs, see the conf param of make_trace_data_requests()
```

Comment:
nit: slightly strange name choice, why not `config`?

Reply:
I actually think I got `traces_source` from a comment of yours on the last PR 😂 

I agree `config` is a better name in hindsight so I will change it to `config`.